### DISCUSSION TITLE
resize tmpfs - tmpfs is not large enough for large gcode file upload

### DISCUSSION THREAD

**Discussion by chawasit** [Guilouz/Creality-Helper-Script-Wiki#552] [Features Request]
I found an 502 bad gateway error when upload 110MB gcode from orca slicer.
Error from moonraker say it has problem with space.
![image](https://github.com/Guilouz/Creality-Helper-Script-Wiki/assets/7046500/d89af369-ba5f-4cfa-8228-5b49ed116cb3)

Moonraker try to write upload file to tmpfs first, then move it later. And, The default tmpfs size is half of a ram, which is 104MB.
![image](https://github.com/Guilouz/Creality-Helper-Script-Wiki/assets/7046500/57d95129-eb26-4458-9948-7e05b634709c)

I try to resize the /tmp to 1GB and its work
`
mount -o remount,size=1G /tmp
`

Hope someone write this to the helper script to permanent resize tmpfs on boot.

some dioscussion related to tmpfs size
#505

**Comment by Guilouz** [comment]
Don't do this, that can cause issue with firmware update!

I have fixed this by relocating Moonraker tmp folder to /usr/data/moonraker/tmp than /tmp who is limited by 98MB.

Replace **S56moonraker_service** file in /etc/init.d folder : https://github.com/Guilouz/Creality-Helper-Script/blob/main/files/services/S56moonraker_service

**Reply by chawasit** [reply]
That is a better fix. Thanks you!
